telemetry/otelgrpc/otelgrpc.go

```diff
@@ -0,0 +1,217 @@
+package otelgrpc
+
+import (
+	"context"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/goto/salt/utils"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/protobuf/proto"
+)
+
+type UnaryParams struct {
+	Start  time.Time
+	Method string
+	Req    any
+	Res    any
+	Err    error
+}
+
+type Meter struct {
+	duration     metric.Int64Histogram
+	requestSize  metric.Int64Histogram
+	responseSize metric.Int64Histogram
+	attributes   []attribute.KeyValue
+}
+
+type MeterOpts struct {
+	meterName string
+}
+
+type Option func(*MeterOpts)
+
+func WithMeterName(meterName string) Option {
+	return func(opts *MeterOpts) {
+		opts.meterName = meterName
+	}
+}
+
+func NewMeter(hostName string, opts ...Option) Meter {
+	meterOpts := &MeterOpts{}
+
+	for _, opt := range opts {
+		opt(meterOpts)
+	}
+
+	meter := otel.Meter(meterOpts.meterName)
```

Comment:
We should have a default value here, but if user pass the meter name with `WithMeterName` , the passed value should be more preferrable

Reply:
added, can you pls check @mabdh